source/documentation/monitoring-an-app/application-metrics.md

```diff
@@ -1,107 +1,121 @@
+run Rails.application
+```
 
-Example:
+If you're running this locally, you'll now be able to query your `/metrics` endpoint and display an output. 
 
-```yaml
-kind: Service
-apiVersion: v1
-metadata:
-  name: my-app-service
-  labels:
-    app: my-app
-spec:
-  selector:
-    app: my-app
-  ports:
-  - protocol: TCP
-    port: 8000
-    targetPort: 8000
-    name: metrics
-  type: NodePort
+```
+curl localhost:3000/metrics
 ```
 
-#### Create a Service-Monitor
-A `ServiceMonitor` is a resource the Prometheus Operator introduces for Kubernetes that describes the set of targets to be monitored by Prometheus
+Build, tag and push your application changes to your code repository and deploy the latest version into your Cloud Platform namespace. Confirm your `/metrics` endpoint is now accessible from your url. 
 
-Service Monitor Architecture
-![Image of Service-Monitor Architecture](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/service-monitor-arch.png)
+```
+curl https://myapp.cloud-platform/metrics
+```
 
+#### Add Service endpoint and ServiceMonitor
 
-Example:
+We need to expose the metrics endpoint with a `Service` and tell the Cloud Platform Prometheus to scrape the endpoint with `ServiceMonitor` object in Kubernetes. In this example, we're using the same port to expose both our application and metrics endpoint so we'll need to query our existing `Service` for the current port name and number. However, if you're exposing a different port you'll need to either amend your current `Service` or create a new one. 
 
-```yaml
-apiVersion: monitoring.coreos.com/v1
-kind: ServiceMonitor
-metadata:
-  name: my-app
-spec:
-  selector:
-    matchLabels:
-      app: my-app
-  endpoints:
-  - port: metrics
-    interval: 15s
-```
+Let's find out our current port name and number by running:
 
-More detailed information about service-monitors can be found [here](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/running-exporters.md)
+```
+kubectl get svc -n <namespace>
+```
 
-#### NetworkPolicy for Monitoring Namespace
+Create and apply your service monitor `<application>-serviceMonitor.yaml`, as below:
 
+```yaml
+   apiVersion: monitoring.coreos.com/v1
+   kind: ServiceMonitor
+   metadata:
+     name: rails-app-service
+   spec:
+     selector:
+       matchLabels:
+         app: rails-app-service
+     endpoints:
+     - port: http # this is the port name you grabbed from your running service
+       interval: 15s
+   ```
+
+This will tell Prometheus to go and scrape that endpoint every 15 seconds and store any exposed metrics.
+
+#### Add a NetworkPolicy resource
+ 
 By default, all connections from outside a namespace are blocked, therefore a network policy is required for the `monitoring` namespace to be able to connect into an application namespace to scrape the metrics endpoint.
```

Comment:
I think it would be clearer if we add something like "The prometheus server is in the 'monitoring' namespace, but by default, any network connections from outside your application's namespace will be blocked. So, to allow prometheus to scrape your application's /metrics endpoint, we need to add a network policy to allow connections from the monitoring namespace.

Reply:
Agreed. Added your suggested change.